arangod/IResearch/IResearchDataStore.h

```diff
@@ -63,20 +63,16 @@ using LinkLock = AsyncLinkHandle::Value;
 ////////////////////////////////////////////////////////////////////////////////
 struct IResearchTrxState final : public TransactionState::Cookie {
   // prevent data-store deallocation (lock @ AsyncSelf)
-  LinkLock _linkLock;  // should be first field to destroy last
+  LinkLock _linkLock;                   // should be first field to destroy last
   irs::IndexWriter::Transaction _ctx;
   PrimaryKeyFilterContainer _removals;  // list of document removals
-  bool _wasCommit = false;
 
   IResearchTrxState(LinkLock&& linkLock, irs::IndexWriter& writer) noexcept
       : _linkLock{std::move(linkLock)}, _ctx{writer.GetBatch()} {}
 
   ~IResearchTrxState() final {
-    if (!_wasCommit) {
-      _removals.clear();
-      _ctx.Reset();
-    }
-    TRI_ASSERT(_removals.empty());
+    _removals.clear();
+    _ctx.Abort();
```

Comment:
I kept old order, but probably more logical opposite.
Abort and only then clear